src/backend/distributed/commands/collation.c

```diff
@@ -588,9 +590,12 @@ PostprocessDefineCollationStmt(Node *node, const char *queryString)
 	ObjectAddress collationAddress =
 		DefineCollationStmtObjectAddress(node, false);
 
-	EnsureDependenciesExistOnAllNodes(&collationAddress);
+	if (IsObjectDistributed(&collationAddress))
+	{
+		EnsureCoordinator();
+	}
 
-	MarkObjectDistributed(&collationAddress);
+	EnsureDependenciesExistOnAllNodes(&collationAddress);
```

Comment:
Since we were using EnsureDependenciesExistOnAllNodes before the PR as well, we don't switch to it. We've set markDistributed to ture for collation definition in distributed_object_ops so we can mark it distributed after creating on worker nodes.

Reply:
sorry, I didn't see we already have `EnsureDependenciesExistOnAllNodes` before